subprojects/plugins/src/main/java/org/gradle/api/tasks/SourceSetOutput.java

```diff
@@ -54,33 +54,30 @@
  *
  * <pre class='autoTested'>
  * plugins {
- *     id 'java'
+ *   id 'java'
  * }
  *
- * def generatedResources = "$buildDir/generated-resources/main"
+ * def generateResourcesTask = tasks.register("generate-resources", GenerateResourcesTask) {
+ *   resourcesDir.set(file("$buildDir/generated-resources/main"))
```

Comment:
can you change this around to use `layout.buildDirectory.dir(...)`?

Reply:
Done